From 139a05ba7c5bd07ee267429790ff1139f428fc67 Mon Sep 17 00:00:00 2001
From: Bruce Ashfield <bruce.ashfield@windriver.com>
Date: Mon, 25 Mar 2013 12:28:06 -0700
Subject: [PATCH] mips: fix signal.h SA_RESTORER build breakage

-stable commit cdef9602 [signal: always clear sa_restorer on execve]
assumes the presence of sa_restorer in the signal_handler structure.
This is not the case for MIPS o32, although the architecture does
defined SA_RESTORER. To fix this, we add the structure to the mips
local copy of signal.h.

Signed-off-by: Bruce Ashfield <bruce.ashfield@windriver.com>
---
 arch/mips/include/uapi/asm/signal.h |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/mips/include/uapi/asm/signal.h b/arch/mips/include/uapi/asm/signal.h
index 770732c..a88343a 100644
--- a/arch/mips/include/uapi/asm/signal.h
+++ b/arch/mips/include/uapi/asm/signal.h
@@ -99,6 +99,9 @@ typedef unsigned long old_sigset_t;		/* at least 32 bits */
 struct sigaction {
 	unsigned int	sa_flags;
 	__sighandler_t	sa_handler;
+#ifdef SA_RESTORER
+	__sigrestore_t sa_restorer;
+#endif 
 	sigset_t	sa_mask;
 };
 
-- 
1.7.10.4

